pkg/odo/cli/logs/logs.go

```diff
@@ -173,18 +200,20 @@ func getUniqueContainerName(name string, uniqueNames map[string]struct{}) string
 }
 
 // printLogs prints the logs of the containers with container name prefixed to the log message
-func printLogs(containerName string, rd io.ReadCloser, out io.Writer) error {
-	color.Set(log.ColorPicker())
-	defer color.Unset()
+func printLogs(containerName string, rd io.ReadCloser, out io.Writer, colour color.Attribute, mu *sync.Mutex) error {
 	scanner := bufio.NewScanner(rd)
 	scanner.Split(bufio.ScanLines)
 
 	for scanner.Scan() {
 		line := scanner.Text()
+		mu.Lock()
+		color.Set(colour)
 		_, err := fmt.Fprintln(out, containerName+": "+line)
 		if err != nil {
 			return err
```

Comment:
If there is an error, the mutex will never be unlocked, no? Wouldn't that be an issue?

Reply:
Probably not, because if we hit error, the execution stops anyway. But it's a bad design, so I'm making the change.